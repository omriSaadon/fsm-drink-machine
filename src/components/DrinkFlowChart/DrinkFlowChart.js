import React, { useEffect, useState } from 'react';
import ReactFlow from 'react-flow-renderer';
import { useStore } from 'store/FsmStore';
import { getMachineStatus } from 'store/fsm_reducer';
import { FLOW_INITIAL_ELEMENTS } from 'constants/index';
import classes from './DrinkFlowChart.scss';

const DrinkFlowChart = () => {
  const [elements, setElements] = useState(FLOW_INITIAL_ELEMENTS);
  const [state] = useStore();

  const drinkMachineStatus = getMachineStatus(state);

  useEffect(() => {
    const newElements = elements.map((element) => {
      if (element.id === drinkMachineStatus) {
        return { ...element, style: { background: '#D6D5E6' } };
      }
      return { ...element, style: { ...element.style, background: null } };
    });
    setElements(newElements);
  }, [state]);

  return (
    <div className={classes.flow}>
      <ReactFlow
        elements={elements}
        zoomOnScroll={false}
        nodesDraggable={false}
        elementsSelectable={false}
        zoomOnDoubleClick={false}
        paneMoveable={false}
        nodesConnectable={false}
      />
    </div>
  );
};

export default DrinkFlowChart;
