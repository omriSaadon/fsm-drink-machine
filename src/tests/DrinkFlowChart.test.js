import React from 'react';
import DrinkFlowChart from 'components/DrinkFlowChart/DrinkFlowChart';
import { machineRender } from './helpers/machineHelper';

jest.mock('react-flow-renderer', () => ({
  default: () => <div>Mock test flow</div>,
  __esModule: true,
}));

test('Verify snapshot', () => {
  const { asFragment } = machineRender(<DrinkFlowChart />);
  expect(asFragment()).toMatchSnapshot();
});
