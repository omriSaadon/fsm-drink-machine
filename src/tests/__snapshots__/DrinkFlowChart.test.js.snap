// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      Mock test flow
    </div>
  </div>
</DocumentFragment>
`;
